adapters/loopme/loopme.go

```diff
@@ -0,0 +1,132 @@
+package loopme
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	reqDatas := make([]*adapters.RequestData, 0, len(request.Imp))
+	for _, imp := range request.Imp {
+		_, err := parseBidderExt(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requestCopy := *request
+		requestCopy.Imp = []openrtb2.Imp{imp}
+		reqJSON, err := json.Marshal(requestCopy)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		headers := http.Header{}
+		headers.Add("Content-Type", "application/json;charset=utf-8")
+		headers.Add("Accept", "application/json")
+		reqDatas = append(reqDatas, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpoint,
+			Body:    reqJSON,
+			Headers: headers,
+			ImpIDs:  openrtb_ext.GetImpIDs(requestCopy.Imp),
+		})
+
+	}
+	return reqDatas, errs
+}
+
+func (a *adapter) MakeBids(bidReq *openrtb2.BidRequest, reqData *adapters.RequestData, respData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(respData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(respData); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(respData.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	if len(bidResp.SeatBid) == 0 || len(bidResp.SeatBid[0].Bid) == 0 {
+		return nil, nil
+	}
+
+	resp := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid[0].Bid))
+	if len(bidResp.Cur) != 0 {
+		resp.Currency = bidResp.Cur
+	}
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			bid := &sb.Bid[i]
+			bidType, err := getBidType(bid)
+			if err != nil {
+				return nil, []error{err}
+			}
+			resp.Bids = append(resp.Bids, &adapters.TypedBid{
+				Bid:     bid,
+				BidType: bidType,
+			})
+		}
+	}
+	return resp, nil
+}
+
+func parseBidderExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpLoopme, error) {
```

Comment:
You don’t need this function because it appears just to be unmarshaling the exts to ensure that they are well formed. This is unnecessary because bidder param validation following the rules specified in the bidder-params JSON file occurs upstream and if that bidder param validation fails, the adapter is never called. Therefore it is safe for the adapter to assume that the bidder params are valid.

Reply:
In such case I don't need ext unmarshaling test cases at all?